datafusion/sqllogictest/test_files/string/string_literal.slt

```diff
@@ -829,3 +838,737 @@ SELECT
+    ('%'), ('\%'), ('\\%'),
+    ('%%'), ('\%%'),
+    ('_'), ('\_'), ('\\_'),
+    ('__'), ('\__'), ('\\__'),
+    ('abc'), ('a_c'), ('a\_c'),
+    ('%abc'), ('\%abc')
+)
+SELECT l.a, r.a, l.a LIKE r.a
+FROM data l CROSS JOIN data r
+----
+NULL NULL NULL
+NULL (empty) NULL
+NULL \ NULL
+NULL \\ NULL
+NULL \\\ NULL
+NULL \\\\ NULL
+NULL a NULL
+NULL \a NULL
+NULL \\a NULL
+NULL % NULL
+NULL \% NULL
+NULL \\% NULL
+NULL %% NULL
+NULL \%% NULL
+NULL _ NULL
+NULL \_ NULL
+NULL \\_ NULL
+NULL __ NULL
+NULL \__ NULL
+NULL \\__ NULL
+NULL abc NULL
+NULL a_c NULL
+NULL a\_c NULL
+NULL %abc NULL
+NULL \%abc NULL
+(empty) NULL NULL
+(empty) (empty) true
+(empty) \ false
+(empty) \\ false
+(empty) \\\ false
+(empty) \\\\ false
+(empty) a false
+(empty) \a false
+(empty) \\a false
+(empty) % true
+(empty) \% false
+(empty) \\% false
+(empty) %% true
+(empty) \%% false
+(empty) _ false
+(empty) \_ false
+(empty) \\_ false
+(empty) __ false
+(empty) \__ false
+(empty) \\__ false
+(empty) abc false
+(empty) a_c false
+(empty) a\_c false
+(empty) %abc false
+(empty) \%abc false
+\ NULL NULL
+\ (empty) false
+\ \ true
+\ \\ false
+\ \\\ false
+\ \\\\ false
+\ a false
+\ \a false
+\ \\a false
+\ % true
+\ \% false
+\ \\% false
+\ %% true
+\ \%% false
+\ _ true
+\ \_ false
+\ \\_ false
+\ __ false
+\ \__ false
+\ \\__ false
+\ abc false
+\ a_c false
+\ a\_c false
+\ %abc false
+\ \%abc false
+\\ NULL NULL
+\\ (empty) false
+\\ \ false
+\\ \\ true
+\\ \\\ false
+\\ \\\\ false
+\\ a false
+\\ \a false
+\\ \\a false
+\\ % true
+\\ \% false
+\\ \\% false
+\\ %% true
+\\ \%% false
+\\ _ false
+\\ \_ false
+\\ \\_ false
+\\ __ true
+\\ \__ false
+\\ \\__ false
+\\ abc false
+\\ a_c false
+\\ a\_c false
+\\ %abc false
+\\ \%abc false
+\\\ NULL NULL
+\\\ (empty) false
+\\\ \ false
+\\\ \\ false
+\\\ \\\ true
+\\\ \\\\ false
+\\\ a false
+\\\ \a false
+\\\ \\a false
+\\\ % true
+\\\ \% false
+\\\ \\% false
+\\\ %% true
+\\\ \%% false
+\\\ _ false
+\\\ \_ false
+\\\ \\_ false
+\\\ __ false
+\\\ \__ false
+\\\ \\__ false
+\\\ abc false
+\\\ a_c false
+\\\ a\_c false
+\\\ %abc false
+\\\ \%abc false
+\\\\ NULL NULL
+\\\\ (empty) false
+\\\\ \ false
+\\\\ \\ false
+\\\\ \\\ false
+\\\\ \\\\ true
+\\\\ a false
+\\\\ \a false
+\\\\ \\a false
+\\\\ % true
+\\\\ \% false
+\\\\ \\% false
+\\\\ %% true
+\\\\ \%% false
+\\\\ _ false
+\\\\ \_ false
+\\\\ \\_ false
+\\\\ __ false
+\\\\ \__ false
+\\\\ \\__ false
+\\\\ abc false
+\\\\ a_c false
+\\\\ a\_c false
+\\\\ %abc false
+\\\\ \%abc false
+a NULL NULL
+a (empty) false
+a \ false
+a \\ false
+a \\\ false
+a \\\\ false
+a a true
+a \a false
+a \\a false
+a % true
+a \% false
+a \\% false
+a %% true
+a \%% false
+a _ true
+a \_ false
+a \\_ false
+a __ false
+a \__ false
+a \\__ false
+a abc false
+a a_c false
+a a\_c false
+a %abc false
+a \%abc false
+\a NULL NULL
+\a (empty) false
+\a \ false
+\a \\ false
+\a \\\ false
+\a \\\\ false
+\a a false
+\a \a true
+\a \\a false
+\a % true
+\a \% false
+\a \\% false
```

Comment:
This one looks like a bug.
Given `\` is an escape character, `\\` should match single backslash character `\` of the input,
and `%` should match the rest (`a`). The result should be `true` then